Services/Certificate/classes/class.ilCertificate.php

```diff
@@ -343,7 +343,7 @@ public function createArchiveDirectory()
 		$type = ilObject::_lookupType($this->objectId);
 		$certificateId = $this->objectId;
 
-		$dir = $this->certificatePath . time() . "__" . IL_INST_ID . "__" . $type . "__" . $certificateId . "__certificate/";
+		$dir = CLIENT_WEB_DIR . $this->certificatePath . time() . "__" . IL_INST_ID . "__" . $type . "__" . $certificateId . "__certificate/";
```

Comment:
You are absolutely right :+1:  Instead of using the global `CLIENT_WEB_DIR` and `ilUtil` we should consider the current approach of using `ILIAS\Filesystem\Filesystem` instead. In this case e.g. `$DIC->filesystem()->web();`

What do you think @bheyser  @mjansenDatabay ?

Reply:
Find ich gut (!) Das würde dann in etwa so aussehen wenn ich mich nicht irre:

`	public function createArchiveDirectory()
	{
		global $DIC; /* @var ILIAS\DI\Container $DIC */
		$webStorage = $DIC->filesystem()->web();
		
		$type = ilObject::_lookupType($this->objectId);
		$certificateId = $this->objectId;

		$dir = $this->certificatePath . time() . "__" . IL_INST_ID . "__" . $type . "__" . $certificateId . "__certificate/";
		if( !$webStorage->hasDir($dir) )
		{
			$webStorage->createDir($dir);
		}
		return $webStorage->getAbsolutePath($dir);
	}
`

Haken: Die Methode getAbsolutePath() habe ich gerade frei erfunden ;-)
Aber die hiesige Methode muss ja damit alles weitere auch funktioniert den absoluten Pfad zurück geben.

@mjansenDatabay Hast Du eine Idee? Selbst am Ende zur Not doch wieder die Konstante zu verwenden find ich nicht verwerflich. In Zukunft könnte das ja noch umgebaut werden, wenn dann an allen Stellen der Filesystem Service verwendet wird.